tests/flask_integration/test_accountant_dashboard_view.py

```diff
@@ -1,30 +1,29 @@
-from .flask import ViewTestCase
+from typing import Optional
 
+from parameterized import parameterized
 
-class AuthenticatedAsAccountantTests(ViewTestCase):
-    def setUp(self) -> None:
-        super().setUp()
-        self.accountant = self.login_accountant()
-
-    def test_get_200_response(self) -> None:
-        response = self.client.get("/accountant/dashboard")
-        self.assertEqual(response.status_code, 200)
+from .flask import LogInUser, ViewTestCase
 
 
-class UnauthenticatedTests(ViewTestCase):
+class AuthTests(ViewTestCase):
     def setUp(self) -> None:
         super().setUp()
-
-    def test_does_not_return_200_response(self) -> None:
-        response = self.client.get("/accountant/dashboard")
-        self.assertNotEqual(response.status_code, 200)
-
-
-class AuthenticatedAsMemberTests(ViewTestCase):
-    def setUp(self) -> None:
-        super().setUp()
-        self.login_member()
-
-    def test_does_not_return_200_response(self) -> None:
-        response = self.client.get("/accountant/dashboard")
-        self.assertNotEqual(response.status_code, 200)
+        self.url = "/accountant/dashboard"
+
+    @parameterized.expand(
+        [
+            (LogInUser.accountant, 200),
```

Comment:
One could argue that we don't need to exercise all 4 cases for this route since this behavior is not even part of the view code but instead of the authentication logic. I would IMO suffice to test this only for an accountant and then again for a non-accountant. This would already demonstrate that we use the proper authentication decorator in our flask code.

Those two remaining test cases should then also be separated into individual test cases since the name of the current test is really not descriptive.

Reply:
With the context from [above](https://github.com/arbeitszeit/arbeitszeitapp/pull/722#discussion_r1199751511) this comment is obsolete.